### PR TITLE
Add while and accompanying features

### LINKED ISSUE #154: While loops and accompanying features
## While loops
An important feature missing from N for too long has been the `while` loop. While it may not have too many useful functionalities, now that the `for` loop has started iterating through lists, it is important that this feature is to be added. There are many ways of implementing it but I will currently suggest this:
```js
import SystemIO
let name = ""
while (~validName(name)) {
  var name = SystemIO.inp("Please input a name")
}
```
This would allow for easy usage of the `while` loop that is familiar to those that do not know N that well, as it imitates the way that other programming languages use `while` loops.
## Continue
`continue` is a keyword already reserved in the fixing of #80 along with `while` and `break`. It would be added in a way that one can just call it like a simple statement as such:
```js
for (i in range(0, 10, 1)) {
  if (i == 1) {
    continue
  }
  print(i)
}
```
This would be a simple, and familiar way of using `continue` to most people. It should be easy to implement as it is similar to `return` but with less error checking.
## Break
`break` is a very important keyword that for some reason has been left unimplemented for a long time, yet it reserved as a keyword when #80 was fixed. The way `break` would work is that it would be called in a similar way to `continue` except instead of continuing to the next iteration, it would break out of the loop entirely, this could also be used in `if` statements as it might be very helpful. There are some foreseeable issues with scoping break as in the situation.
```js
if (1 == 1) {
  if (true) {
    break
  }
  if (true) {
   print("hi")
  }
}
```
it may be that the user does not want to exit out of the first if statement of vice versa, in this situation it is hard to accurately judge the scope of the `break`.

### PR DESCRIPTION
**Pull request by Ashvin-Ranjan**
closes #154 

## TODO
- [x] Running 
  - [x] `while`
  - [x] `break`
  - [x] `continue`

### CODE REVIEW

File: python/scope.py [hunk 1]
```diff
@@ -1036,10 +1048,30 @@ async def eval_command(self, tree):
...
+        elif command.data == "while":
+            var, code = command.children
+            val = await self.eval_expr(var)
+            while val:
+                scope = self.new_scope()
+
+
+                exit, value = await scope.eval_command(code)
+                if exit == True: # exit can be a string
+                    return True, value
```
Comment (SheepTester) [inline python/scope.py:1064]: Hmm, wouldn't this make `break` return out of the function? Also, you can avoid the comparison with `True` by moving this after `if exit == "continue":`

**Comment by SheepTester** [comment]
Can you make sure this works?

```ts
let wow = [a: int] -> int {
  while (true) {
    break
  }
  return a + 1
}

print(wow(3)) // should print 4
```

File: python/scope.py [hunk 2]
```diff
@@ -1048,10 +1048,12 @@ async def eval_command(self, tree):
 
                 scope.assign_to_pattern(pattern, i, certain=True)
                 exit, value = await scope.eval_command(code)
-                if exit == True: # exit can be a string
-                    return True, value
                 if exit == "continue":
                     continue
+                if exit:
+                    if value == None:
+                        return False, None
```
Comment (SheepTester) [inline python/scope.py:1055]: Hmm, wouldn't a better solution be to do what you did with continue and have it return `"break"`?

Comment (Ashvin-Ranjan) [reply python/scope.py]: Added

File: python/scope.py [hunk 3]
```diff
@@ -1050,9 +1050,9 @@ async def eval_command(self, tree):
                 exit, value = await scope.eval_command(code)
                 if exit == "continue":
                     continue
+                if exit == "break":
+                    return False, None
```
Comment (SheepTester) [inline python/scope.py:1054]: Awesome, `break` would work too

In the future, if we add labels, we can use the second item for labels

**Pull request merged**